### PR TITLE
SPACE TRAIN EMERGENCY SHUTTLE

### PR DESCRIPTION
**Pull request by ghost**
CLICKBAIT

:cl: Armhulen
add: Buy a space train today, chucklenuts!
/:cl:

It's box, but smaller, and longer. It'll be fun to have syndicates fight their way to the cockpit, so lets see how that goes down.

image

### CODE REVIEW

**Comment by Armhulen** [comment]
pictures when i please

**Comment by nicbn** [comment]
Pics

**Comment by nicbn** [comment]
Need pics

**Comment by nicbn** [comment]
Post pics

**Comment by nicbn** [comment]
Send pics

**Comment by nicbn** [comment]
Pics please

**Comment by QualityVan** [comment]
pictures where

**Comment by SpaceManiac** [comment]
>not map merged
>no pics
>doesn't compile

**Comment by Armhulen** [comment]
:anger:

**Comment by Armhulen** [comment]
OH FUCK MAP MERGE

**Comment by Armhulen** [comment]
SON OF A BITCH

File: code/datums/shuttles.dm [hunk 1]
```diff
@@ -80,6 +80,13 @@
 	admin_notes = "Includes a small variety of weapons. And bears. Only captain-access can release the bears. Bears won't smash the windows themselves, but they can escape if someone lets them."
 	credit_cost = 5000 // While the shuttle is rusted and poorly maintained, trained bears are costly.
 
+/datum/map_template/shuttle/emergency/train
+	suffix = "train"
+	name = "The Emergency Escape Train
```
Comment (nicbn) [inline code/datums/shuttles.dm]: CLOSE THAT FUCKING QUOTES

File: _maps/shuttles/emergency_train.dmm [hunk 2]
```diff
@@ -0,0 +1,79 @@
+"a" = (/turf/template_noop,/area/template_noop)
```
Comment (nicbn) [inline _maps/shuttles/emergency_train.dmm]: MAPMERGE THAT FUCKING MAP

**Review by nicbn — CHANGES_REQUESTED** [review]

File: code/datums/shuttles.dm (hunk 1 above)
Comment (Armhulen) [reply code/datums/shuttles.dm]: whoops

**Comment by Armhulen** [comment]
image

**Comment by ShiggyDiggyDo** [comment]
50f 1

**Comment by Mark9013100** [comment]
THAT'S NOT THE SNOW PIERCER REEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE

**Comment by KorPhaeron** [comment]
Needs tiny fans in airlocks so you can move on the connectors without spacewind

**Comment by Armhulen** [comment]
that's what i wanted but qbopper said it was bad design

**Comment by KorPhaeron** [comment]
qbopper is a redditor

**Comment by Armhulen** [comment]
tiny fans it is

**Comment by Okand37** [comment]
How will the airlocks line up with the station's docks?

**Comment by Armhulen** [comment]
kind of not really but somewhat from what i predict i haven't seen it

**Comment by GunHog** [comment]
You made a highly irregular shuttle and did not bother to test it? :/

**Comment by Armhulen** [comment]
it's just a shuttle, and the places i put the doors SHOULD be connecting. i'm just not completely sure

**Comment by Okand37** [comment]
If you'd like a reference, this is the standard docking airlock setup for most stations. Granted, shuttles like asteroidstation's emergency or birdboat's emergency don't properly connect, either, so it isn't too far fetched... but its good for atleast 1-2 airlocks to line up.

**Comment by Armhulen** [comment]
two airlocks meet up we're good.

**Comment by GunHog** [comment]
Does the shuttle smash any parts of a station such as Meta?

**Comment by Armhulen** [comment]
most definitely

File: _maps/shuttles/emergency_train.dmm [hunk 3]
```diff
@@ -0,0 +1,518 @@
...
+	dir = 4
+	},
+/turf/open/floor/mineral/titanium,
+/area/shuttle/escape)
+"j" = (
+/turf/open/floor/mineral/titanium/blue,
+/area/shuttle/escape)
+"k" = (
+/obj/machinery/computer/card{
+	tag = "icon-computer (WEST)";
```
Comment (ShizCalev) [inline _maps/shuttles/emergency_train.dmm]: Dirty var

File: _maps/shuttles/emergency_train.dmm [hunk 4]
```diff
@@ -0,0 +1,518 @@
...
+	dir = 8
+	},
+/obj/machinery/light/small{
+	dir = 4
+	},
+/turf/open/floor/mineral/titanium,
+/area/shuttle/escape)
+"M" = (
+/obj/machinery/light/small{
+	tag = "icon-bulb1 (NORTH)";
```
Comment (ShizCalev) [inline _maps/shuttles/emergency_train.dmm]: Var

File: _maps/shuttles/emergency_train.dmm [hunk 5]
```diff
@@ -0,0 +1,518 @@
...
+	pixel_y = -30
+	},
+/turf/open/floor/mineral/titanium/blue,
+/area/shuttle/escape)
+"y" = (
+/obj/machinery/door/airlock/titanium{
+	name = "Emergency Shuttle Airlock"
+	},
+/obj/docking_port/mobile/emergency{
+	name = "Train emergency shuttle";
```
Comment (ShizCalev) [inline _maps/shuttles/emergency_train.dmm]: Not a fan of mixed casing

File: _maps/shuttles/emergency_train.dmm [hunk 6]
```diff
@@ -0,0 +1,518 @@
...
+	},
+/obj/machinery/light/small{
+	dir = 4
+	},
+/turf/open/floor/mineral/titanium,
+/area/shuttle/escape)
+"M" = (
+/obj/machinery/light/small{
+	tag = "icon-bulb1 (NORTH)";
+	icon_state = "bulb1";
```
Comment (ShizCalev) [inline _maps/shuttles/emergency_train.dmm]: Var

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: _maps/shuttles/emergency_train.dmm [hunk 7]
```diff
@@ -0,0 +1,518 @@
...
+/area/shuttle/escape)
+"I" = (
+/obj/machinery/light/small{
+	dir = 4
+	},
+/turf/open/floor/mineral/plastitanium/brig,
+/area/shuttle/escape)
+"J" = (
+/obj/machinery/light/small{
+	tag = "icon-bulb1 (NORTH)";
```
Comment (ShizCalev) [inline _maps/shuttles/emergency_train.dmm]: Var

File: _maps/shuttles/emergency_train.dmm [hunk 8]
```diff
@@ -0,0 +1,518 @@
...
+	icon_state = "bulb1";
```
Comment (ShizCalev) [inline _maps/shuttles/emergency_train.dmm]: Var

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by Qbopper** [comment]
@KorPhaeron using tiny fans means you've failed as a mapper tbqh

but arm didn't implement any of the suggestions made to resolve the problem without using tiny fans

so tiny fans it is!!!

**Comment by Armhulen** [comment]
@Qbopper what, glass? are you lame?

@ShizCalev i didn't do this i just placed the blocks! how to i fix the var problem

**Comment by epochayur** [comment]
this is dumb but I really like it

**Comment by Qbopper** [comment]
no, not glass windows, you apparently didn't see what else I said

the pr is already up though so who cares

**Comment by Armhulen** [comment]
@Qbopper enlighten me please :(

**Comment by DaxDupont** [comment]
Trains are cool

**Comment by Anonmare** [comment]
I like trains

**Comment by calzilla1** [comment]
MOAR SHUTTLES FOR THE SHUTTLE GOD

**Comment by FantasticFwoosh** [comment]
More of a concern without using the tiny fans is basically even if space wind is not a threat to yourself passing between carriages if you have something like magboots & a firesuit, space wind will push items & people behind you increasing the risk that you will be displaced and fall off.

Also incredibly easy to troll people by spilling a bottle of water for a wet puddle on the carriage connectors, instantly spacing them as the wind pull them away while they are lying down on their ass.

**Comment by Armhulen** [comment]
if you're too scared to go on the connectors, don't go on the connectors?

**Comment by FantasticFwoosh** [comment]
Its a train @Armhulen unless you ride in the front or have 2 sets of doors (1 safe passage between carriages & the risky connectors)  you can't pass from the front to the back of the shuttle and basically station crew will buy this train-ship especially to screw with traitors because sec won't be able to get to the back or drag people out of the carriages into holding once it departs.

- Just kill everyone else in the 3rd/4th carriage during radio silence, escape not in custody easy because security are too scared to drag you back to the front of the train.

You can afford to make it as wide as a normal shuttle so there is more room for architecture as the connectors don't matter on the first & second carriages with double doors (which is better design)

**Comment by Qbopper** [comment]
@Armhulen "just don't use the cool gimmick part of the gimmick shuttle" is a lame defense lol

**Comment by Buggy123** [comment]
Currently, there is:
A shuttle who gibs anyone waiting to escape if they're inattentive or incapacitated.
A shuttle consisting of a cramped room surrounding a highly radioactive and instant-dusting supermatter. With emitter beams firing at it constantly.
A shuttle which will burst into a raging inferno at the slightest provocation.
A shuttle which contains bears.
A shuttle which leads to some sort of hell dimension IDK I've never actually seen it used.
And a DIY shuttle which is left incomplete and depressurized 90% of the time anyway.

All things considered, adding "a shuttle where you have to go through 2 tiles of vacuum to get to another room" really doesn't seem that bad.

**Comment by GunHog** [comment]
You make a good point...Maybe we could feature creep this one where if the connecting tiles are destroyed, the rear section(s) will disconnect and stall out in space (or fall to Lavaland)!

**Comment by Qbopper** [comment]
make it so there's passengers on the train so you can yell about it being a train robbery

**Comment by Armhulen** [comment]
Bopper if they're too afraid to use the gimmick of the shuttle they don't have to

**Comment by Armhulen** [comment]
don't merge this yet btw

**Comment by calzilla1** [comment]
Maybe add glass panes to the sides of the connectors?

**Comment by DoctoreMad** [comment]
Wait you have to go through space to get to the top of the shuttle?

**Comment by GunHog** [comment]
Yes, you pass through space to get the next section in the shuttle. My guess is that is designed to simulate those old movies and games where characters are exposed to the outside air while passing through train cars.

**Comment by Armhulen** [comment]
look i gotta finish krakens and changelings and mallets so this one will sit in the backseat

you guys got your disco meme shuttle anyway

**Pull request closed without merging**